File: 0703-kth-largest-element-in-a-stream/0703-kth-largest-element-in-a-stream.java
class KthLargest {
    PriorityQueue<Integer> minHeap;
    int k;

    public KthLargest(int k, int[] nums) {
        minHeap = new PriorityQueue<>();
        this.k = k;

        for(var num : nums){
            minHeap.offer(num);
            if(minHeap.size() > k) minHeap.poll();
        }
    }
    
    public int add(int val) {
        minHeap.offer(val);
        if(minHeap.size() > k) minHeap.poll();
        return minHeap.peek();
    }
}

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest obj = new KthLargest(k, nums);
 * int param_1 = obj.add(val);
 */
 

File: 0703-kth-largest-element-in-a-stream/0703-kth-largest-element-in-a-stream.cs
public class KthLargest {
    PriorityQueue<int, int> minHeap;
    int k;

    public KthLargest(int k, int[] nums) {
        minHeap = new(nums.Select(x => (x, x)));
        this.k = k;

        while(minHeap.Count > k) minHeap.Dequeue();
    }
    
    public int Add(int val) {
        minHeap.Enqueue(val, val);
        if(minHeap.Count > k) minHeap.Dequeue();
        return minHeap.Peek();
    }
}

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest obj = new KthLargest(k, nums);
 * int param_1 = obj.Add(val);
 */